pallets/pools/src/lib.rs

```diff
@@ -994,6 +958,97 @@ pub mod pallet {
 			T::Time::now().as_secs()
 		}
 
+		pub(crate) fn do_update_invest_order(
+			who: &T::AccountId,
+			pool: &mut PoolDetailsOf<T>,
+			order: &mut UserOrderOf<T>,
+			amount: T::Balance,
+			pool_id: T::PoolId,
+			tranche_id: T::TrancheId,
+		) -> DispatchResult {
+			let mut outstanding = &mut pool
+				.tranches
+				.get_mut(tranche_id.into())
+				.ok_or(Error::<T>::InvalidData)?
+				.outstanding_invest_orders;
+			let pool_account = PoolLocator { pool_id }.into_account();
+
+			let (send, recv, transfer_amount) = Self::update_order_amount(
+				who,
+				&pool_account,
+				&mut order.invest,
+				amount,
+				&mut outstanding,
+			)?;
+
+			order.epoch = pool.current_epoch;
+			T::Tokens::transfer(pool.currency, send, recv, transfer_amount)
+		}
+
+		pub(crate) fn do_update_redeem_order(
+			who: &T::AccountId,
+			pool: &mut PoolDetailsOf<T>,
+			order: &mut UserOrderOf<T>,
+			amount: T::Balance,
+			pool_id: T::PoolId,
+			tranche_id: T::TrancheId,
+		) -> DispatchResult {
+			let currency = T::TrancheToken::tranche_token(pool_id, tranche_id);
+			let mut outstanding = &mut pool
+				.tranches
+				.get_mut(tranche_id.into())
+				.ok_or(Error::<T>::InvalidData)?
+				.outstanding_redeem_orders;
+			let pool_account = PoolLocator { pool_id }.into_account();
+
+			let (send, recv, transfer_amount) = Self::update_order_amount(
+				who,
+				&pool_account,
+				&mut order.redeem,
+				amount,
+				&mut outstanding,
+			)?;
+
+			order.epoch = pool.current_epoch;
+			T::Tokens::transfer(currency, send, recv, transfer_amount)
+		}
+
+		fn update_order_amount<'a>(
+			who: &'a T::AccountId,
+			pool: &'a T::AccountId,
+			order: &mut T::Balance,
+			update_order: T::Balance,
+			pool_orders: &mut T::Balance,
+		) -> Result<(&'a T::AccountId, &'a T::AccountId, T::Balance), DispatchError> {
+			ensure!(*order != update_order, Error::<T>::InvalidData);
```

Comment:
Is there a reason to fail here instead of just making this a no-op with an OK result? Either way no state got updated, but when I think about us maybe wanting to integrate pools with other logic in the future, idempotence is a nice property

Reply:
My reasoning was, that I don`t want events triggered although nothing was updated. @offerijns WDYT? I am happy with both approaches.

Reply:
I changed it to no-op. Was the previous state anyways. We can change it later on if we think the event is an issue.

Reply:
It's a bit tricky to change, as we rely on the fact, that a transfer of zero is a no-op. Which it is, but It looks a bit weird. Will do anyways.

Reply:
So changing this to a no-op then.

Reply:
Actually, from a state perspective, this is not a no-op. As the root of the state changes. I think I will stay with erroring out here.